crates/sparrow-instructions/src/columnar_value.rs

```diff
@@ -62,6 +62,13 @@ impl ColumnarValue {
         ArcRef::new(array).try_map(|a| downcast_map_array(a))
     }
 
+    /// Specialized version of `array_ref` that downcasts the array to a
+    /// list array.
+    pub fn list_array(&self) -> anyhow::Result<ArcRef<dyn Array, ListArray>> {
+        let array = self.array_ref()?;
+        ArcRef::new(array).try_map(|a| downcast_list_array(a))
```

Comment:
Instead of `downcast_list_array(a)` can do `a.as_list_opt()` (see https://docs.rs/arrow/latest/arrow/array/trait.AsArray.html#tymethod.as_list_opt)

Reply:
Removed `ArcRef` and using `as_list()` as per discussion